src/libcmd/installables.cc

```diff
@@ -662,7 +664,9 @@ std::tuple<std::string, FlakeRef, InstallableValue::DerivationInfo> InstallableF
             priority = aPriority->getInt();
     }
 
-    if (outputsToInstall.empty() || std::get_if<AllOutputs>(&extendedOutputsSpec)) {
+    auto * outputsSpec = std::get_if<ExtendedOutputsSpec::Explicit>(&extendedOutputsSpec);
```

Comment:
Idem.

Reply:
I rewrote this logic to hopefully be much easier to read. There was some stuff that just seemed redundant? I deleted it but do check.